arangod/Agency/Agent.cpp

```diff
@@ -909,17 +927,18 @@ futures::Future<query_t> Agent::poll(
       VPackObjectBuilder r(builder.get());
       builder->add(VPackValue("result"));
       VPackObjectBuilder r2(builder.get());
-      builder->add("commitIndex", VPackValue(_commitIndex));
+      builder->add("commitIndex", VPackValue(_commitIndex.load(std::memory_order_relaxed)));
       builder->add("firstIndex", VPackValue(0));
       builder->add(VPackValue("readDB"));
       _readDB.get("", *builder, true);
-    } else if (index <= _commitIndex) {   // deliver immediately all logs since index
+    } else if (index <= _commitIndex.load(std::memory_order_relaxed)) {
+      // deliver immediately all logs since index
       builder = std::make_shared<VPackBuilder>();
       VPackObjectBuilder r(builder.get());
       builder->add(VPackValue("result"));
       VPackObjectBuilder r2(builder.get());
-      logs = _state.get(index, _commitIndex);
-      builder->add("commitIndex", VPackValue(_commitIndex));
+      logs = _state.get(index, _commitIndex.load(std::memory_order_relaxed));
+      builder->add("commitIndex", VPackValue(_commitIndex.load(std::memory_order_relaxed)));
```

Comment:
There is again a potential race here in case `_commitIndex` changes between the `if` statement and the next two usage of it.

Reply:
_commitIndex is only changed under _outputLock. the atomicity was only introduced for reads